terminatorlib/util.py

```diff
@@ -407,20 +407,13 @@ def update_config_to_cell_height(filename):
 
 def get_flatpak_args(args, envv, cwd):
     """Contruct args to be executed via flatpak-spawn"""
-    import json
     flatpak_args = None
-    shell = [args[0]]
     env_args = ['--env={}'.format(env) for env in envv]
     flatpak_spawn = [
         "flatpak-spawn", "--host", "--watch-bus", "--forward-fd=1",
         "--forward-fd=2", "--directory={}".format(cwd)
     ]
-    if len(args) == 2:
-        flatpak_args = flatpak_spawn + env_args + shell
-
-    # Support -x, -e, custom commands etc.
-    if len(args) > 2:
-        flatpak_args = flatpak_spawn + env_args + args[1:]
+    flatpak_args = flatpak_spawn + env_args + args
```

Comment:
@mattrose this is simplifying the flatpak args. Now it will only prepend flatpak-spawn on front of existing args. I have this if statement there before to avoid running shell with '-c' as flatpak-spawn could just execute commands without it.

With this change flatpak-spawn will always execute `'/bin/bash', '-l', '-c', 'your custom shell command'` when you specify custom command as well as "run as logon shell"

I'm not sure why terminator args are constructed in a way that you always need to run `$shell -c customcommand` when running terminator with things like `-x` and `-e` or custom command in terminator settings.

Do I understand correctly that this is required when using `self.vte.spawn_sync()` ?

Anyway, with this change it has a very simple job which is to take whatever terminator had in args and add it on top of `flatpak-spawn --host --env... args`

This means that when you have no custom settings specified for custom shell instead of executing `/bin/bash` it will execute `flatpak-spawn --host /bin/bash`

When you enable `login_shell = True`:
Instead of running `/bin/bash -l` it will run `flatpak-spawn --host /bin/bash -l`

When you set custom command in settings to /bin/zsh:
Instead of running `/bin/bash -c /bin/zsh` it will run `flatpak-spawn --host /bin/bash -c /bin/zsh`

And with custom command set as well as login shell set : 
Instead of running `/bin/bash -l -c /bin/zsh` it will run `flatpak-spawn --host /bin/bash -l -c /bin/zsh` 

I believe this should make it in line and working with the current setup with all options working.

Perhaps you could consider switching rest of the code to just use `self.vte.spawn_async()` instead of current `self.vte.spawn_sync()` . I saw somewhere that spawn_sync() is deprecated.

Reply:
Thanks, I'll take another look.  The first pass looked like it was just grabbing the user's shell and passing that to vte.spawn_async(), but it's possible I could have missed something.  I'll take another look now. 


The reason terminator doesn't use spawn_async is because it's a relatively new call that isn't supported by Ubuntu 18.04, which is still supported and relatively popular.   What I should do is fix it so that if spawn_async is available then it'll use that and fall back to spawn_sync if they're using an older vte.

Reply:
> The reason terminator doesn't use spawn_async is because it's a relatively new call that isn't supported by Ubuntu 18.04, which is still supported and relatively popular. What I should do is fix it so that if spawn_async is available then it'll use that and fall back to spawn_sync if they're using an older vte.

I see. That makes sense. I guess this problem will be going away with flatpak.

Reply:
Actually It looks like I still need to tweak it making it "simple" to work with all these extra command breaks the normal behaviour with default config :| 

`Terminal::spawn_child: Forking shell: "/bin/bash" with args: ['flatpak-spawn', '--host', '--watch-bus', '--forward-fd=1', '--forward-fd=2', '--directory=/var/home/jaydoubleu', '--env=TERM=xterm-256color', '--env=COLORTERM=truecolor', '--env=PWD=/var/home/jaydoubleu', '--env=TERMINATOR_UUID=urn:uuid:f84be701-e3d3-4d54-862e-dd9d4371dab9', '--env=TERMINATOR_DBUS_NAME=net.tenshu.Terminator25ef4b219e3b005583550f2b0f9f990c3', '--env=TERMINATOR_DBUS_PATH=/net/tenshu/Terminator2', '/bin/bash', '/bin/bash'] via flatpak-spawn`

While the spawn_sync() was working with args like this ` '/bin/bash', '/bin/bash'` the spawn_async() is not happy about it at all.

Going to look into it further